sys/include/test_utils/result_output.h

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (C) 2021 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    test_utils_result_output Test result output
+ * @ingroup     sys
+ * @brief       Utility function for abstraction of test result output format
+ *
+ * The TURO module provides an abstraction layer allowing salient data to be
+ * provided for tests independent of format or medium. The intention is to have
+ * a test that expects some data, for example, reading some registers,
+ * output the results in a know way, for example json.  This should help
+ * keeping the test results stable and not lock anyone into a perticular
+ * format. If JSON is too heavy all tests using this can be swapped out for
+ * something lighter, for example CBOR. Then the tests should not have to be
+ * adapted. There can also be python layers that coordinate the output results,
+ * ideally done with riotctrl.
+ *
+ * Some of the design decisions include:
+ * - ability to flush immediately to prevent the need for large buffers
+ * - selectable output format based on USEMODULE
```

Comment:
It would be really helpful to document which modules are available. The base functionality and the different outputs, maybe a table or so that could be extended easily?

Reply:
I think this was not addressed yet too.

Reply:
I think it has with the doxygen update

Reply:
or not? One sec.

Reply:
Ya, I guess doxygen doesn't pick up the stuff from the .c files of the implementation.  I will look into it.

Reply:
https://8598-7370241-gh.circle-artifacts.com/0/doc/group__test__utils__result__output.html

Reply:
There we go!

Reply:
Mh I would actually like the modules to be listed here as well. Something like:

```suggestion
 * - selectable output format based on `USEMODULE`:
 *   + `test_utils_result_output_check`: @ref test_utils_result_output_check
 *   + `test_utils_result_output_json`: @ref test_utils_result_output_json
 *   + `test_utils_result_output_txt`: @ref test_utils_result_output_txt
```

Reply:
done